.drone.yml

```
@@ -904,6 +904,143 @@ steps:
+      - git checkout $GODL_COMMIT
+      - cd $TELEPORT_DIR
+      - git clone https://github.com/gravitational/teleport.git .
+      - git checkout ${DRONE_TAG:-$DRONE_COMMIT}
+      # fetch enterprise submodules
+      # suppressing the newline on the end of the private key makes git operations fail on MacOS
+      # with an error like 'Load key "/path/.ssh/id_rsa": invalid format'
+      - mkdir -m 0700 $WORKSPACE_DIR/.ssh && echo "$GITHUB_PRIVATE_KEY" > $WORKSPACE_DIR/.ssh/id_rsa && chmod 600 $WORKSPACE_DIR/.ssh/id_rsa
+      - ssh-keyscan -H github.com > $WORKSPACE_DIR/.ssh/known_hosts 2>/dev/null
+      - chmod 600 $WORKSPACE_DIR/.ssh/known_hosts
+      - GIT_SSH_COMMAND='ssh -i $WORKSPACE_DIR/.ssh/id_rsa -o UserKnownHostsFile=$WORKSPACE_DIR/.ssh/known_hosts -F /dev/null' git submodule update --init e
+      # this is allowed to fail because pre-4.3 Teleport versions don't use the webassets submodule
+      - GIT_SSH_COMMAND='ssh -i $WORKSPACE_DIR/.ssh/id_rsa -o UserKnownHostsFile=$WORKSPACE_DIR/.ssh/known_hosts -F /dev/null' git submodule update --init --recursive webassets || true
+      - rm -rf $WORKSPACE_DIR/.ssh
+      - mkdir -p $WORKSPACE_DIR/go/cache
+
+  - name: Build the boringcrypto Go builder
+    environment:
+      GODL_DIR: /tmp/push-build-darwin-amd64-fips/go/src/github.com/gravitational/godl
+      BORINGCRYPTO_BRANCH: dev.boringcrypto.go1.15
+    commands:
+      - set -u
+      - cd $GODL_DIR
+      - mkdir build
+      - go build -o $GODL_DIR/build/$BORINGCRYPTO_BRANCH $BORINGCRYPTO_BRANCH/main.go
+
+  - name: Pull boringcrypto build tarball from S3
+    environment:
+      AWS_S3_BUCKET:
+        from_secret: AWS_S3_BUCKET
+      AWS_ACCESS_KEY_ID:
+        from_secret: AWS_ACCESS_KEY_ID
+      AWS_SECRET_ACCESS_KEY:
+        from_secret: AWS_SECRET_ACCESS_KEY
+      AWS_REGION: us-west-2
+      BORINGCRYPTO_BRANCH: dev.boringcrypto.go1.15
+      GODL_DIR: /tmp/push-build-darwin-amd64-fips/go/src/github.com/gravitational/godl
+    commands:
+      - set -u
+      - mkdir -p $HOME/sdk/dev.boringcrypto.go1.15
+      - cd $HOME/sdk/dev.boringcrypto.go1.15
+      - aws s3 cp s3://$AWS_S3_BUCKET/ci/go/$BORINGCRYPTO_BRANCH-darwin-amd64.tar.gz .
+      - aws s3 cp s3://$AWS_S3_BUCKET/ci/go/$BORINGCRYPTO_BRANCH-darwin-amd64.tar.gz.sha256 .
+      - $GODL_DIR/build/$BORINGCRYPTO_BRANCH download
+
+  - name: Build Mac artifacts
+    environment:
+      GOPATH: /tmp/push-build-darwin-amd64-fips/go
+      GOCACHE: /tmp/push-build-darwin-amd64-fips/go/cache
+      # use custom Go compiler
+      GO_BINARY: /tmp/push-build-darwin-amd64-fips/go/src/github.com/gravitational/godl/build/dev.boringcrypto.go1.15
+      OS: darwin
+      ARCH: amd64
+      WORKSPACE_DIR: /tmp/push-build-darwin-amd64-fips
+      FIPS: "yes"
+    commands:
+      - set -u
+      - cd $WORKSPACE_DIR/go/src/github.com/gravitational/teleport
+      - make clean release OS=$OS ARCH=$ARCH FIPS=$FIPS GO_BINARY=$GO_BINARY
```

We don't provide/create FIPS builds for OSS, so it might be better to run something like this:

```suggestion
      - make -C e clean release VERSION=$VERSION OS=$OS ARCH=$ARCH FIPS=$FIPS GO_BINARY=$GO_BINARY
```

You'd need to set the `VERSION` variable using similar logic to what we do here: https://github.com/gravitational/teleport/blob/master/.drone.yml#L1748-L1769